cns/metric/configsnapshot.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018 Microsoft. All rights reserved.
+// MIT License
+
+package metric
+
+import (
+	"context"
+	"crypto/md5" //nolint:gosec // used for checksum
+	"encoding/json"
+	"time"
+
+	"github.com/Azure/azure-container-networking/aitelemetry"
+	"github.com/Azure/azure-container-networking/cns/configuration"
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/pkg/errors"
+)
+
+// SendCNSConfigSnapshot emits CNS config periodically
+func SendCNSConfigSnapshot(ctx context.Context, config *configuration.CNSConfig) {
+	ticker := time.NewTicker(time.Minute * time.Duration(config.TelemetrySettings.ConfigSnapshotIntervalInMins))
+	defer ticker.Stop()
+
+	event, err := createCNSConfigSnapshotEvent(config)
+	if err != nil {
+		logger.Errorf("[Azure CNS] SendCNSConfigSnapshot: %v", err)
+		return
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			logger.LogEvent(event)
+		}
+	}
+}
+
+func createCNSConfigSnapshotEvent(config *configuration.CNSConfig) (aitelemetry.Event, error) {
+	bb, err := json.Marshal(config) //nolint:musttag // no tag needed for config
+	if err != nil {
+		return aitelemetry.Event{}, errors.Wrap(err, "failed to marshal config")
+	}
+
+	cs := md5.Sum(bb) //nolint:gosec // used for checksum
+	csStr := string(cs[:])
```

Comment:
why do we need a checksum?

Reply:
Yeah, in theory you'd be able to filter lines where these get emitted and easily pinpoint when the config changed, which is probably of most interest.  Doing that precise filtering would also probably filter out all of the CNS startup fanfare that happens in the logs too, so I think it remains useful even if the config is immutable while CNS is running.

Reply:
we have a separate table for this type of telemetry (snapshots) so no need to worry about logs at least